mat64/triangular_test.go

```diff
@@ -79,61 +132,41 @@ func (s *S) TestTriDenseCopy(c *check.C) {
 		size := rand.Intn(100)
 		r, err := randDense(size, 0.9, rand.NormFloat64)
 		if size == 0 {
-			c.Check(err, check.Equals, ErrZeroLength)
+			if err != ErrZeroLength {
+				c.Fatalf("expected error %v: got: %v", ErrZeroLength, err)
+			}
 			continue
 		}
-		c.Assert(err, check.Equals, nil)
+		if err != nil {
+			c.Fatalf("unexpected error: %v", err)
+		}
 
 		u := NewTriDense(size, true, nil)
 		l := NewTriDense(size, false, nil)
 
-		u.Copy(r)
-		l.Copy(r)
-		for m := 0; m < size; m++ {
-			for n := 0; n < size; n++ {
-				e := r.At(m, n)
-				switch {
-				case m < n: // Upper triangular matrix.
-					c.Check(u.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-				case m == n: // Diagonal matrix.
-					c.Check(u.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-					c.Check(l.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-				case m < n: // Lower triangular matrix.
-					c.Check(l.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-				}
-			}
-		}
-
-		u.Copy((*basicVectorer)(r))
-		l.Copy((*basicVectorer)(r))
-		for m := 0; m < size; m++ {
-			for n := 0; n < size; n++ {
-				e := r.At(m, n)
-				switch {
-				case m < n: // Upper triangular matrix.
-					c.Check(u.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-				case m == n: // Diagonal matrix.
-					c.Check(u.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-					c.Check(l.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-				case m < n: // Lower triangular matrix.
-					c.Check(l.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-				}
-			}
-		}
-
-		u.Copy((*basicMatrix)(r))
-		l.Copy((*basicMatrix)(r))
-		for m := 0; m < size; m++ {
-			for n := 0; n < size; n++ {
-				e := r.At(m, n)
-				switch {
-				case m < n: // Upper triangular matrix.
-					c.Check(u.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-				case m == n: // Diagonal matrix.
-					c.Check(u.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-					c.Check(l.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
-				case m < n: // Lower triangular matrix.
-					c.Check(l.At(m, n), check.Equals, e, check.Commentf("Test #%d At(%d, %d)", i, m, n))
+		for _, typ := range []Matrix{r, (*basicVectorer)(r), (*basicMatrix)(r)} {
+			u.Copy(typ)
```

Comment:
Seems like there should be a zeroing of u and l (or something) so that the copy is fresh for each new type.